docs/src/main/asciidoc/security-proactive-authentication.adoc

```diff
@@ -110,6 +110,51 @@ However, some CDI beans may not be ready for use.
 One exception to this solution is a situation when application endpoints are secured with the xref:security-authorize-web-endpoints-reference.adoc#authorization-using-configuration[Authorization using configuration].
 For more information, see the xref:security-authorize-web-endpoints-reference.adoc#policy-active-cdi-request-context[Inject RequestScoped beans into HttpSecurityPolicy] section of the "Authorization of Web endpoints" guide for more information.
 
+=== Make `SecurityIdentity` available in newly activated CDI request context
+
+There are infrequent situations where Quarkus cannot populate CDI request context with current `SecurityIdentity` because you activated CDI request context before Quarkus could prepare it.
+Most common example is when you register a route on the xref:reactive-routes.adoc#using-vertx-web-router[Vert.x Web Router] like in the example below:
+
+[source,java]
+----
+package io.quarkus.it.security;
+
+public class RouterObserver {
+
+    public void route(@Observes Router router, UserInformation userInformation) {
+        router.route("/user-info").handler(event -> event.response().end(userInformation.getPrincipalName()));
+    }
+
+}
+----
+
+The `UserInformation` bean in the example can look like this:
+
+[source,java]
+----
+package io.quarkus.it.security;
+
+@ApplicationScoped
+public class UserInformation {
+
+    @Inject
+    SecurityIdentity identity;
+
+    @ActivateRequestContext
+    String getPrincipalName() {
+        return identity.getPrincipal().getName();
+    }
+
+}
+----
+
+For this example to work, you need to activate `SecurityIdentity` propagation with the Vert.x duplicated context local data:
+
+[source,properties]
+----
+quarkus.http.auth.always-propagate-security-identity=true
+----
+
```

Comment:
I think it would be worth adding a note here clarifying again, in addition to the intro above, that users must only do it when registering Vert.x handlers directly on the vert.x router where no `RoutingContext` is available and that it is not needed for the routes registered declaratively with `@Route` as well as JAX-RS endpoints.